raiden-ts/src/transport/utils.ts

```diff
@@ -33,33 +34,52 @@ export const getPresences$: (action$: Observable<RaidenAction>) => Observable<Pr
 );
 
 /**
- * Stringify a caps mapping
+ * Stringify a caps mapping to a caps url
+ * E.g.'mxc://raiden.network/cap?k1=true&k2=v2&k2=v3&k4=null&k5=123'
  *
  * @param caps - Capabilities object/mapping
  * @returns stringified version of caps
  */
 export function stringifyCaps(caps: Caps): string {
-  return Object.entries(caps)
-    .filter(([, v]) => typeof v !== 'boolean' || v)
-    .map(([k, v]) => (typeof v === 'boolean' ? k : `${k}="${v}"`))
-    .join(',');
+  const url = new URL('mxc://raiden.network/cap');
+  // URLSearchParams.append can handle all primitives
+  const appendParam = (key: string, value: string | number | boolean | null) =>
```

Comment:
Isn't the value type of `CapsPrimitive`?

Reply:
yes, but `CapsPrimitive` is a codec, not a type.. I wanted to avoid exporting it just because of this, but thinking better, I'll do it. Thanks